### ISSUE TITLE
Clear database unlock password field after set period of time

### ISSUE THREAD

**Issue by dadosch** [keepassxreboot/keepassxc#4162]
[TIP]:  # ( Provide a general summary of the issue in the title above ^^ )
[TIP]:  # ( DO NOT include screenshots of your actual database! )

## Expected Behavior
[NOTE]: # ( Tell us what you expected to happen )
The field should be cleared, as it is possible to click the "view password" afterwards to reveil the password. If you enter your password only with a small typo, then leave keepassxc open, it might be possible for a 3rd party to know your password (for example if you misspelled a word which is in a dictionary)

## Current Behavior
[NOTE]: # ( Tell us what actually happens )
the database unlock password field is not cleared after a failed attempt to unlock the database.

## Possible Solution
[NOTE]: # ( Not required, but suggest a fix/reason for the bug )
the field should be cleared, as it was the case in previous version of KeePassXC

## Steps to Reproduce
1. enter something into database unlock password field
2. click enter
3. click the eye to reveil what you have entered

## Context
[NOTE]: # ( How has this issue affected you? What unique circumstances do you have? )

## Debug Info
KeePassXC - Version 2.5.2
Revision: 62cda9d

Qt 5.14.0
Diagnosemodus ist deaktiviert.

Betriebssystem: Arch Linux
CPU-Architektur: x86_64
Kernel: linux 5.4.6-arch3-1

Aktivierte Erweiterungen:
- Auto-Type
- Browser-Integration
- SSH-Agent
- KeeShare (bestätigtes und unbestätigtes Teilen)
- YubiKey
- Secret-Service-Integration

Kryptographische Bibliotheken:
 libgcrypt 1.8.5

**Comment by droidmonkey** [comment]
We changed the behavior to select the password so it can be retyped easily. Clearing the password would piss off people with really long master passwords that mistype one character. I do not think this specific request should be implemented. I am in favor of auto clearing the password field after a set amount of inactivity, when the window is minimized, etc.

**Comment by dadosch** [comment]
> would piss off people with really long master passwords that mistype one character.
I disagree. I have a very long master password, but I couldn't write it down without thinking which keys I would press (it is basically just the movements of my fingers which I memorized, so I wouldn't be able to identify the typo if I saw the password) so it is faster to retype the whole password.


Maybe this could be a setting, so that people can adjust it to their likes.
